### ISSUE TITLE
Raw to MP3 conversion

### ISSUE THREAD

**Issue by genabotpub** [chrisstaite/lameenc#7]
Hello,

I am new to audio manipulation. In my project I have to convert a raw audio file into an mp3 one.

The raw file (attached az zip)  is recorded with the following characteristics:
`sampleRate = 48000,  sampleSizeInBits = 16,  channels = 1,  signed = true,  bigEndian = true`
[in.zip](https://github.com/chrisstaite/lameenc/files/5764935/in.zip)

I tried to implement the potential solution as described in the attached file.
[raw_to_mp3.txt](https://github.com/chrisstaite/lameenc/files/5764927/raw_to_mp3.txt)

Unfortunately, the resultant mp3 includes only noise.

Would appreciate any advise.

Thank you,
gen

**Comment by chrisstaite** [comment]
There's a few reasons that it's not working for you and a few places where your Python can be improved.  LAME requires your input to be little endian, so for starters you'll need to swap the byte order round of your data.  Using this trick works: https://stackoverflow.com/questions/36096292/efficient-way-to-swap-bytes-in-python/36096506.  Secondly, your input data is MONO, so you'll need to set your channels to 1.  That's enough to get your encoding working.

However, there's a few Python improvements that you may be interested in.  The context manager of the "with" command automatically closes files, so there's no need to call close explicitly.  The scope of variables in Python is defined by the function, so there's no need to pre-declare variables.  Finally, you can return within a with removing the need to have a variable at all.

Here's my version of your script that works for me with your input.

```import lameenc
import os


def main():
    dirName = os.path.dirname(os.path.realpath(__file__))
    myArr = get_raw_audio(dirName)
    # Swap from big-endian to little-endian
    byteswapped = bytearray(len(myArr))
    byteswapped[0::2] = myArr[1::2]
    byteswapped[1::2] = myArr[0::2]

    encoder = lameenc.Encoder()

    encoder.set_bit_rate(256)
    encoder.set_in_sample_rate(48000)
    encoder.set_channels(1)  # Needs to match the input channels
    encoder.set_quality(2)  # 2-highest, 7-fastest

    with open(dirName + "/out.mp3", "wb") as newFile:
        newFile.write(encoder.encode(bytes(byteswapped)))
        newFile.write(encoder.flush())


def get_raw_audio(dirName):
    with open(dirName + "/in.raw", "rb") as binaryfile:
        return binaryfile.read()


if __name__ == "__main__":
    main()
```

In fixing this, I have improved the mono handling in lamenc and fixed a deprecation warning for Python 3.8.  So the latest version should be slightly better too.

**Comment by genabotpub** [comment]
Chris,

Thank you very much for such a comprehensive response. I really appreciate the time you took to address my question.

Your code works in my environment as well.

Also, thank you for the feedback regarding Python - I just started to learn it.

Best regards,
gen
